Instance check opcodes

vlazzarini opened the pull request:
This PR addresses https://github.com/csound/csound/issues/2093, adding opcodes to check for active and releasing instances.

vlazzarini committed:
added instance check opcodes

gesellkammer commented:
"isreleasing" seems to do somewhat the same as the "release" opcode. Could these be unified?

vlazzarini commented:
yes, you are right - I forgot about that. We probably don't need a new name, just overload release to take an instance parameter, what do you reckon?

vlazzarini committed:
working on changes to op names

vlazzarini committed:
Merge branch 'develop' into feature/isactive

vlazzarini committed:
changed opcode names and adjusted tests - added boolean version

vlazzarini commented:
I have now changed names to `release(inst:Instr)` and `active(inst:Instr)` for consistency. Also added `res:B = release()` as an overload to the existing `res:k = release()`. Also placed all opcodes together in the same file.

gesellkammer commented:
I did like the previous names better... Particularly "release" sounds
confusingly imperative, like asking the instr to stop. Would it be an idea
to keep the previous names and make the old "release" an alias?

vlazzarini ***@***.***> schrieb am Fr., 25. Juli 2025, 15:35:

> *vlazzarini* left a comment (csound/csound#2194)
> <https://github.com/csound/csound/pull/2194#issuecomment-3117848333>
>
> I have now changed names to release(inst:Instr) and active(inst:Instr)
> for consistency. Also added res:B = release() as an overload to the
> existing res:k = release(). Also placed all opcodes together in the same
> file.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/csound/csound/pull/2194#issuecomment-3117848333>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAT3NERU53IVV5WYGBX3L33KIXB7AVCNFSM6AAAAACBNFZHZOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCMJXHA2DQMZTGM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

vlazzarini commented:
that's also possible.

vlazzarini committed:
reverted name changes

vlazzarini commented:
ok, so now we have various flavours of `isreleasing()`, and `isactive()` gets its name back,

```
instr 1
 xtratim 0.1
 actv:b = isactive(this)
 if actv then
   prints "this is running\n"
 else
   exitnow(-1)
 endif

 rels:B = isreleasing(this)
 if rels then
   printks "releasing this\n", 1
   printk  0.1, times:k()
 endif
 
 rels:B = isreleasing()
 if rels then
   printks "releasing - boolean\n", 1
   printk  0.1, times:k()
 endif

 rels1:k = isreleasing()
 if rels1 > 0 then
   printks "releasing - original\n", 1
   printk  0.1, times:k()
 endif
endin
```
